### ISSUE TITLE
How to change ScrollToTop behavior?

### ISSUE THREAD

**Issue by kodekrendel** [nuxt/nuxt.js#521]
I read documentation, but still can't handle it. 

I want to change default ScrollToTop behavior. How to make ScrollToTop **after** navigating to the link, not to ScrollToTop in current page? Can somebody help me with this?

**Comment by alexchopin** [comment]
Can you explain more about what you are trying to do ?

**Comment by kodekrendel** [comment]
@alexchopin when I click link, scroll jumps to top of current page and then link is loaded with next page. What I want is scroll not to jump to top on current page (where I click the link), and just load the next page (where scroll will be on top).

**Comment by alexchopin** [comment]
If you are moving to another page, what is the issue with the scroll behavior? 
I mean why are you caring about the current position if you want another one on the next page?

**Comment by kodekrendel** [comment]
@alexchopin it's about UX, it can be confusing to reader. For example, I have a long article, user clicks link on bottom expecting that he will move on next page (with smooth lovely transitions that nuxt provide out of the box), but he rapidly jumps on top of the same page and only then he got transition effect and loading next page. To many action, I think. 

Anyway, if this is hard to implement, then ok, it's not that critical. Just wonder and want to try without  ScrollToTop, how it will look.

**Comment by ausir0726** [comment]
@kodekrendel  I know what ware you talking about .

I think you see that behavior it because you setting to long transition time . 
( .page-enter-active, .page-leave-active )
if you just use default 0.5s , that will be nice , and scrollToTop behavior will improve .
your demo site , the page you leave is disappear too slow , so you see it move top first then disappear , then page enter ...

so , if you want new page ( route ) enter , then scroll to top , that will some weird behavior .

TL;DR
think about ...
if you have two pages ,  /a-page  /b-page 
a-page  have  1000px  height
b-page have 1300px height

when a-page on 900 height jump to b-page 
if as you wish ..
what will be (a) 900px -> (b) -> 900px -> 0px
ok ... seens good ...
if a-page 900px  , but b-page only 600px 
so , that will be 900px -> 600px -> 0px
that will be always show new page on button ...
if you have lazyload ? will load button img ? weird ....

----
any way you , should shorten .page-leave-active transition
maybe less than 1s

**Comment by kodekrendel** [comment]
@ausir0726 I doesn't change transition time from default, I just add timing function:

```
.page-enter-active, .page-leave-active
{
  transition: opacity .5s;
  transition-timing-function: cubic-bezier(0.8,0,1,0);
}
```

**Comment by ausir0726** [comment]
ummm .....

it's look like scroll to top then add page-leave-active ...
so ...

maybe need find source code ... QQa

in fact
my personal feelings that add page-leave-active class >> scroll to tpo >> page-enter-active class
will more appropriate ...

**Comment by kodekrendel** [comment]
That what I think also, scroll to top should be after leaving the page.

As for source, I already look at it, but I lack of skills - https://nuxtjs.org/api/pages-scrolltotop/

**Comment by ausir0726** [comment]
not , maybe more deeper in vue-router .....

**Comment by kodekrendel** [comment]
@ausir0726 I guess it is?

```
export default new Router({
  mode: '<%= router.mode %>',
  base: '<%= router.base %>',
  linkActiveClass: '<%= router.linkActiveClass %>',
  scrollBehavior,
  routes: [
<%= _routes %>
  ]
})
```

**Comment by kodekrendel** [comment]
What if add delay to scrollBehavior? 500ms

**Comment by Atinux** [comment]
I've seems this behavior indeed, I believe it's a vue-router issue, I will investigate about it.

**Comment by max-schu** [comment]
temporary workarround:

~plugins/router-guards.js
```
import router from '~router'

router.afterEach((to, from) => {
  console.log(to)
  // set timeout time to time of your transition
  setTimeout(() => {
    // dunno why but conditional helps to speed up build time…
    if (document) document.body.scrollTop = 0
  }, 500)
})
```

nuxt.config.js
```
  router: {
    scrollBehavior: function (to, from, savedPosition) {
      return { x: false, y: false }
    }
  },
```

**Comment by kodekrendel** [comment]
@max-schu I see no effect

**Comment by kodekrendel** [comment]
Oh, sorry, it works!

**Comment by kodekrendel** [comment]
Very nice, much smoother UX. Thank you @max-schu

**Comment by kodekrendel** [comment]
Having little bug, since Scrollbar blinking after we route to page where Scrollbar even not have to be present (small viewport). If I increase time in router-guards.js then this bug disappears, but other appears - blinking of Footer when route.

**Comment by jvandenaardweg** [comment]
The issue still exists. In the latest version a scroll to top still happens right before the route change. I have 0 "scrollToTop" properties set.

**Comment by kodekrendel** [comment]
@jvandenaardweg what do you mean "still exist"? There were no official fix.

**Comment by jvandenaardweg** [comment]
I thought there was, because it got closed since 0.10.7. My bad then.

I've worked around this one by using nuxt-child, and setting scollToTop to false.

It's a bit annoying scrollToTop is a default behaviour BEFORE switching routes though

**Comment by Atinux** [comment]
It's a problem of vue-router directly, see https://github.com/vuejs/vue-router/issues/1263

**Comment by kodekrendel** [comment]
@jvandenaardweg have you tried fix above that @max-schu offers? It's works fine for me (few minor bugs, but ok)

**Comment by jvandenaardweg** [comment]
Yeah i tried, but doesnt work when you rely on fetch to prefetch data before changing routes :-)

**Comment by kodekrendel** [comment]
@jvandenaardweg not sure I understand. You mean dynamic child-routes? Just checked - working for me.

**Comment by jvandenaardweg** [comment]
Well, the workaround relies on a timeout. This is a fixed time. I'm prefetching data when changing routes. So first getting my data from an API and then display it. This way you can't be sure it's taking a fixed amount of time.

**Comment by kodekrendel** [comment]
@ I am also fetching the data with Axios.

**Comment by max-schu** [comment]
should be possible to just get the promise instead of timeout – didn't try it tho

**Comment by jvandenaardweg** [comment]
Btw, this is the ticket at the vue-router repo about the same problem: https://github.com/vuejs/vue-router/issues/1373

Let's hope for a fix fast :-)

**Comment by kodekrendel** [comment]
@jvandenaardweg yep, that's it. Nice to know this issue is noticed.
